### ISSUE TITLE
Iterative result-set

### ISSUE THREAD

**Issue by Seddryck** [Seddryck/NBi#623]
This new feature is tackling use-cases where you need to perform many times unions of result-sets defined with queries having a parameter evolving over the result-set' executions. Another example would be a directory having one files named based on the month of data that they contain (Data_2020_JAN.csv, Data_2020_FEB.csv, Data_2020_MAR.csv ...) and you must execute a union of all these files. You can surely create a long list of static unions but what will happen if the list of monthly files to concatenate is dynamically evaluated?

``` xml
<result-set>
  <iteration>
    <sequence name="month"  type="dateTime">
      <loop-sentinel seed="2020-01-01" terminal="2020-03-01" step="1 month"/>
    </sequence>
  </iteration>
  <result-set>
    <file>
      <path>~Data_{@month:yyyy}_{@month:MM}.csv</path>
    </file>
  </result-set>
</result-set>
```

**Comment by Seddryck** [comment]
Implemented in [1.23.0-beta.76](https://ci.appveyor.com/project/Seddryck/nbi/builds/37254279/artifacts) or on nuget via ```Update-Package NBi.Framework -version 1.23.0-beta0076```.

**Comment by Seddryck** [comment]
Documentation added in http://www.nbi.io/docs/primitive-result-set/#iterative-result-set
